### PR TITLE
mes-8115-ADI3SearchFix

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed issues with search result for ADI3

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image

### CODE REVIEW

**Review by gittins7 — APPROVED** [review]

File: src/app/pages/view-test-result/components/trainer-details-card/trainer-details-card.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class TrainerDetailsCardComponent {
   category: TestCategory | CategoryCode;
 
   @Input()
-  driverPRNNumber: number;
+  reasonForNoAdviceGiven ?: any;
```
Comment (andrewsetterfield) [inline src/app/pages/view-test-result/components/trainer-details-card/trainer-details-card.ts]: please make this typed

File: src/app/pages/view-test-result/components/vehicle-details-card/vehicle-details-card.ts [hunk 2]
```diff
@@ -185,6 +185,7 @@ export class VehicleDetailsCardComponent {
   getFlattenArray = (data: string[]): string => flattenArray(data);
 
   displayRegistration() {
-    return this.isADI3() || !!this.instructorRegistrationNumber || this.shouldShowDimensions || !!this.vehicleDetails;
```
Comment (andrewsetterfield) [inline src/app/pages/view-test-result/components/vehicle-details-card/vehicle-details-card.ts:188]: inconsistant use of ! for the different values comparedto original.  please review.  also can this be unit tested

File: src/app/pages/view-test-result/components/trainer-details-card/trainer-details-card.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/view-test-result/components/trainer-details-card/trainer-details-card.ts]: fixed in newest commit

File: src/app/pages/view-test-result/components/vehicle-details-card/vehicle-details-card.ts (hunk 2 above)
Comment (RLCorp) [reply src/app/pages/view-test-result/components/vehicle-details-card/vehicle-details-card.ts:188]: comments addressed and unit tests added in new commit

**Review by gittins7 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**